### PR TITLE
ROX-20160: Add RHTAP build for `main`

### PR DESCRIPTION
**Pull request by msugakov**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by msugakov** [comment]
Subscription registration script is coming. I tested and it works for UBI 8.5-8.9 and 9.1-9.3.

File: .rhtap/scripts/subscription-manager-bro.sh [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    for item in "${TARGET_BACKUP_PATHS[@]}"; do
+        if [[ -e "${TARGET_DIR}/${item}" ]]; then
+            tar --append -vf "${TARGET_DIR}"/tmp/restore/backup.tar -C "${TARGET_DIR}" "${item}"
+        fi
+    done
+
+    echo "Registering installer container with subscription manager"
+    subscription-manager register --org="$RED_HAT_ORG_ID" --activationkey="$(cat "${SECRET_PATH}")"
+
+    # It is suggested in the following articles that certain files can be linked to $TARGET_DIR/run/secrets,
```
Comment (tommartensen) [inline .rhtap/scripts/subscription-manager-bro.sh:129]: Missing link?

File: .rhtap/scripts/subscription-manager-bro.sh [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    cp --verbose "/workspace/${RHTAP_SECRET_NAME}/${SECRET_KEY}" "${SECRET_PATH}"
+}
+
+function register {
+    if [[ ! -d "${TARGET_DIR}"/etc ]]; then
+        >&2 echo "Error: Looks like target system is not mounted at ${TARGET_DIR}/etc"
+        exit 5
+    fi
+
+    echo "Backing up original artifacts in $TARGET_DIR"
```
Comment (tommartensen) [inline .rhtap/scripts/subscription-manager-bro.sh:117]: Why is this necessary?

Comment (tommartensen) [reply .rhtap/scripts/subscription-manager-bro.sh:117]: ok, after reading on I see that they will be restored in the cleanup function. A comment would be nice.

File: .rhtap/scripts/subscription-manager-bro.sh (hunk 1 above)
Comment (msugakov) [reply .rhtap/scripts/subscription-manager-bro.sh:132]: Indeed. Added.

File: .rhtap/scripts/subscription-manager-bro.sh (hunk 2 above)
Comment (msugakov) [reply .rhtap/scripts/subscription-manager-bro.sh:120]: Added.

File: ui/Makefile [hunk 3]
```diff
@@ -11,7 +11,8 @@ SOURCES += $(shell find apps -type d \( -name node_modules \) -prune -o -print)
 # tracking package.json files of monorepo packages is the easiest way to discover such situations
 PACKAGE_JSON_FILES := $(shell find . -type d \( -name node_modules \) -prune -false -o -name package.json)
 
-export REACT_APP_ROX_PRODUCT_BRANDING := $(shell $(MAKE) --quiet --no-print-directory -C .. product-branding)
+ROX_PRODUCT_BRANDING ?= $(shell $(MAKE) --quiet --no-print-directory -C .. product-branding)
+export REACT_APP_ROX_PRODUCT_BRANDING := $(ROX_PRODUCT_BRANDING)
```
Comment (msugakov) [inline ui/Makefile:15]: @vjwilson

This makes `REACT_APP_ROX_PRODUCT_BRANDING` configurable via `ROX_PRODUCT_BRANDING` environment variable when the latter one is externally set (`?=` assignment).
Basically, both variables have to be set to the same values and `ROX_PRODUCT_BRANDING` is the one that drives everything.

I made the change because I noticed that the old code which calls Makefile in the repo root... requires Golang and other goodies to be installed. We had no issues with that presumably because UI engineers have Golang (God, I hope not because I forced them with my original dumb change) and downstream builder image also has it.

A builder image in RHTAP doesn't have Golang and I saw problems during the build.

The PR is still draft but I'd appreciate your or someone from UI team eyes on it when it's ready.

Comment (vjwilson) [reply ui/Makefile:15]: ^ @alwayshooin @sachaudh @pedrottimark @dvail @bradr5

Comment (msugakov) [reply ui/Makefile:15]: Factored this in a separate PR https://github.com/stackrox/stackrox/pull/9620

**Comment by msugakov** [comment]
Closing this one as it was split and addressed as separate PRs under the same ROX-20160.
